turbopy/diagnostics.py

```diff
@@ -147,6 +153,11 @@ def initialize(self):
             self.csv = CSVOutputUtility(self._input_data["filename"],
                                         diagnostic_size)
 
+        if "interval" in self._input_data:
+            self.interval = self._input_data["interval"]
+        else:
+            self.interval = self._owner.clock.end_time
```

Comment:
I think I see what you are trying to do with this, but setting the value of `interval` to the `end_time` will give a redundant file write at the end of the simulation. It would be best to avoid that.